src/Catalog.API/Services/CatalogAI.cs

```diff
@@ -1,55 +1,76 @@
-﻿using Azure.AI.OpenAI;
+﻿using System.Diagnostics;
+using Microsoft.SemanticKernel.Embeddings;
 using Pgvector;
 
 namespace eShop.Catalog.API.Services;
 
 public sealed class CatalogAI : ICatalogAI
 {
-    private readonly string _aiEmbeddingModel;
-    private readonly OpenAIClient _openAIClient;
+    private const int EmbeddingDimensions = 384;
+    private readonly ITextEmbeddingGenerationService _embeddingGenerator;
 
     /// <summary>The web host environment.</summary>
     private readonly IWebHostEnvironment _environment;
     /// <summary>Logger for use in AI operations.</summary>
     private readonly ILogger _logger;
 
-    public CatalogAI(IOptions<AIOptions> options, IWebHostEnvironment environment, ILogger<CatalogAI> logger, OpenAIClient openAIClient = null)
+    public CatalogAI(IWebHostEnvironment environment, ILogger<CatalogAI> logger, ITextEmbeddingGenerationService embeddingGenerator = null)
     {
-        var aiOptions = options.Value;
-        _openAIClient = openAIClient;
-        _aiEmbeddingModel = aiOptions.OpenAI.EmbeddingName ?? "text-embedding-ada-002";
-        IsEnabled = _openAIClient != null;
+        _embeddingGenerator = embeddingGenerator;
         _environment = environment;
         _logger = logger;
+    }
+
+    /// <summary>Gets whether the AI system is enabled.</summary>
+    public bool IsEnabled => _embeddingGenerator is not null;
+
+    /// <summary>Gets an embedding vector for the specified catalog item.</summary>
```

Comment:
(nit) Would it be better to have these "summary" docs on the interface and then just use `inheritdoc` here?

That way when in the IDE and using an `ICatalogAI` reference, the docs show up?

Reply:
Sure